src/test/scala/chiseltest/experimental/tests/AssertTest.scala

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiseltest.experimental.tests
+
+import chisel3._
+import chisel3.tester._
+import chiseltest.VerilatorBackendAnnotation
+import org.scalatest.freespec.AnyFreeSpec
+
+class AssertFail extends Module {
+  val io = IO(new Bundle {
+    val input = Input(UInt(2.W))
+  })
+
+  printf("io.input is %d\n", io.input) // prints 3 and then 0
+
+  assert(io.input === 3.U) // fails, apparently on the second cycle
```

Comment:
Well, it shouldn't fail with this fix?

Reply:
Changed comment to say it should not fail